kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -165,46 +166,28 @@ public Transaction build(Engine engine) {
     boolean shouldUpdateProtocol = false;
     Metadata metadata = snapshot.getMetadata();
     Protocol protocol = snapshot.getProtocol();
-    if (tableProperties.isPresent()) {
-      Map<String, String> validatedProperties =
-          TableConfig.validateDeltaProperties(tableProperties.get());
-      Map<String, String> newProperties =
-          metadata.filterOutUnchangedProperties(validatedProperties);
-
-      ColumnMapping.verifyColumnMappingChange(
-          metadata.getConfiguration(), newProperties, isNewTable);
```

Comment:
I'm a bit late to this PR but was this change to do this logic even when table properties are present intentional? I think it'll fail the verifyColumnMappingChange check below when it shouldn't , in the case a user didn't define table properties on a subsequent transaction. 

Unless I'm missing something

Reply:
Basically, TLDR if I'm not mistaken I think we still need this validation only in the condition that `tableProperties.isPresent`

Reply:
https://github.com/delta-io/delta/pull/4197